### PR TITLE
Bring updated datepicker to rg2

### PR DESCRIPTION
**Pull request by JonathanWolfe**
https://github.com/RiotGear/rg-datepicker/pull/12#event-420117396

### CODE REVIEW

File: components/date/rg-date.tag [hunk 1]
```diff
@@ -1,51 +1,92 @@
...
+		this.monthFormat = opts.monthFormat || 'MMMM'
+		this.weekFormat = opts.weekFormat || 'ddd'
+		this.dayFormat = opts.dayFormat || 'DD'
+
+		// Convert the given date to a moment object
+		this.date = moment(opts.date || false);
+
+		// Setup our bouding dates
+		this.beginningBound = opts.beginningBound ? moment(opts.beginningBound) : false;
+		this.endingBound = opts.endingBound ? moment(opts.endingBound) : false;
```
Comment (gregorypratt) [inline components/date/rg-date.tag]: What are these used for? They seem to be defined but unused

Comment (JonathanWolfe) [inline components/date/rg-date.tag]: Whoops, leftovers from when I was making it able to do date ranges.

**Comment by gregorypratt** [comment]
Fancy writing some tests for it?!

**Comment by JonathanWolfe** [comment]
Sure. Any tests you want besides:

- Options Passthrough: https://github.com/RiotGear/rg/blob/master/components/date/rg-date.tag#L57-L68
- Locale: https://github.com/RiotGear/rg/blob/master/components/date/rg-date.tag#L75-L83
- Buffers are Correct: https://github.com/RiotGear/rg/blob/master/components/date/rg-date.tag#L108-L125
- Date Change event fire: https://github.com/RiotGear/rg/blob/master/components/date/rg-date.tag#L140-L146
- Shortcuts Work: https://github.com/RiotGear/rg/blob/master/components/date/rg-date.tag#L148-L177
- Showing/Hiding Work: https://github.com/RiotGear/rg/blob/master/components/date/rg-date.tag#L85-L91

**Comment by gregorypratt** [comment]
That's a good start! If you run the karma tests and check the coverage page to see what else needs doing. 100% is ideal 😉

**Comment by gregorypratt** [comment]
I'm only talking about the date picker though btw!

**Pull request merged**